pkg/catalog/helm/setup.go

```diff
@@ -64,7 +71,12 @@ func CatalogSHA256Hash(catalog *v3.Catalog) string {
 	return hex.EncodeToString(hashBytes[:])
 }
 
-func newCache(catalog *v3.Catalog) *Helm {
+func newCache(catalog *v3.Catalog) (*Helm, error) {
+	sanitizedURL, err := validateURL(catalog.Spec.URL)
+	if err != nil {
+		return nil, err
+	}
+	catalog.Spec.URL = sanitizedURL
```

Comment:
Don't save this on the catalog, just use the `sanitizedURL` when creating the Helm struct

Reply:
Thats what I did at first, but the problem with that approach is that `CatalogSHA256Hash` uses `catalog` with the non-sanitized URL first and therefore won't match. Also this has the benefit that it strips out weird chars from the UX.